src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -188,6 +191,77 @@ func (c controller) PostInstallConfigs(wg *sync.WaitGroup) {
 	c.sendCompleteInstallation(true, "")
 }
 
+func (c controller) UpdateBMHs(wg *sync.WaitGroup) {
+	defer wg.Done()
+	for {
+		time.Sleep(GeneralWaitTimeout)
+		exists, err := c.kc.IsMetalProvisioningExists()
+		if err == nil && exists {
+			c.log.Infof("Provisioning CR exists, no need to update BMHs")
+			return
+		}
+
+		bmhs, err := c.kc.ListBMHs()
+		if err != nil {
+			c.log.WithError(err).Errorf("Failed to BMH hosts")
```

Comment:
failed to list?

Reply:
sure it worth error? maybe warning is enough?

Reply:
error is more visible in the logs, in this case i preferred error, since controller is short-lived

Reply:
u will not have lots of errors till masters join?

Reply:
as long as we can connect to any api server (master or bootstrap), we will be able to list.